src/streamlit_app.py

```diff
@@ -36,97 +38,229 @@
+  - present the JSON query.
+
+### Response Structure
+
+```json
+{{
+  "group_by": ["entity.attribute_name"],
+  "metrics": ["entity.metric_name"],
+  "filters": [], // optional, only add the 'filters' attribute when filtering
+  "order": "ORDER BY \"entity.metric_or_attribute_name\" ASC/DESC LIMIT x" // optional, only when top/bottom/last/first filters applied
+}}
+```
 
 
-## Response Rules
 
-1. IF ABOUT DATA - NO TEXT ONLY THE FUNCTION!
-2. Output text if there something missing or the question is about the schema.
-3. DO NOT CALCULATE NEW THINGS. USE ONLY THE SCHEMA.
-4. ONLY USE ATTRIBUTE OR METRICS THAT EXIST. SEARCH AGAIN. FAIL IF CANT FIND.
-5. IF YOU CAN''T FIND A METRIC, DON''T HELP THE USER TO BUILD IT - FAIL INSTEAD.
-6. IF YOU CAN''T FIND AN ATTRIBUTE - LET THE USER KNOW WITH ERROR RESPONSE
 
-## Dealing with dates
+### Dealing with dates
 
 Use date entity for date comparisons.
 
 Use following Snowflake SQL functions for dates:
 * `CURRENT_DATE` to refer to now
 * `DATE_TRUNC` to get boundaries of a time window, i.e. `DATE_TRUNC(month, CURRENT_DATE())` for month
-* `INTEVAL` to see relative time, i.e last_month is "date.month >= CURRENT_DATE() - INTERVAL ''''1 month''''".
+* `INTEVAL` to see relative time, i.e last_month is "MONTH(date.date) >= CURRENT_DATE() - INTERVAL ''1 month''".
 
-# Your Rules
+### Your Rules
 
-filters:
+### filters:
 * compare one attribute to a given constant value. Can use =,<,>,>=,<=
 * Only use `date.date` to filter on dates
 * all filters will apply
 * number can be compared to numbers `table.attr = 5` or ranges `table.attr >= 3 and table.attr < 10`
-* string can be compared to a value `table.attr = val` or values: `table.attr IN (''''val1'''', ''''val2'''', ...)` or LIKE
+* string can be compared to a value `table.attr = val` or values: `table.attr IN ('val1', 'val', ...)` or LIKE
 * boolean can be true or false `table.attr = true` or `table.attr = false`
 * date can use date Snowflake SQL functions and CURRENT_DATE(), i.e. `YEAR(table.attr) = 2023`
+* when asked for Top, Bottom, Last, first you can use `order` attribute which uses SQL `ORDER BY \"x\" LIMIT y ASC/DESC` syntax
 * do not compare attributes
 
-group_by:
+### attributes (group_by):
 * may choose only from the schema attributes
 * everything is automatically connected - you can use any group you want
 
-metrics:
+### metrics:
 * may choose only from the schema metrics
 
-# Your Schema
+## Your Schema
 
-## Attributes
+### Attributes
 
 {attributes}
 
-OTHER ATTRIBUTES ARE NOT ALLOWED
+**OTHER ATTRIBUTES ARE NOT ALLOWED**
 
-## Metrics
+### Metrics
 
 {metrics}
 
-OTHER METRICS ARE NOT ALLOWED
+**OTHER METRICS ARE NOT ALLOWED**
+
+## Examples
+
+### Successful Queries
+
+#### Example 1: Total Revenue
+
+**User:** "What's the total revenue for Q1 2024 per country?"
+
+**Response:**
+
+**Working on** *What is the total of `sales.revenue` by `operation.country` where `date.quarter` is `Q1 2024`?*
```

Comment:
i like the working on - perhaps we can show it?

Reply:
Fixed